### PR TITLE
Add Google Test 1.8.0

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
EWISOTT. Includes google/googletest#1160.

### CODE REVIEW

File: CMake/External_GTest.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    -DGTest_source:PATH=${fletch_BUILD_PREFIX}/src/GTest
+    -P ${fletch_SOURCE_DIR}/Patches/GTest/Patch.cmake
+  CMAKE_GENERATOR ${gen}
+  CMAKE_ARGS
+    ${COMMON_CMAKE_ARGS}
+  )
+
+set(GTEST_ROOT ${vision-tpl_BUILD_INSTALL_PREFIX} CACHE STRING "")
+
+file(APPEND ${vision-tpl_CONFIG_INPUT} "
```
Comment (dstoup) [inline CMake/External_GTest.cmake]: vision_tpl should change to fletch throughout this file.

File: CMake/External_GTest.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+########################################
+# GTest
+########################################
+set(GTEST_ROOT    @GTEST_ROOT@)
```
Comment (dstoup) [inline CMake/External_GTest.cmake]: Fletch has a new pattern for writing out this file. See one of the simple cases, e.g. ExternalZLib.cmake. In short, it looks something like...

file(APPEND ${fletch_CONFIG_INPUT} "
set(GTEST_ROOT \$\{fletch_ROOT\})
set(fletch_ENABLED_GTEST_ROOT TRUE)
")

File: Patches/GTest/Patch.cmake [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#+
+# This file is called as CMake -P script for the patch step of
+# External_GTest.cmake.
+# GTest_patch and GTest_source are defined on the command line along with the
+# call.
+#-
+
+message("Patching GTest")
```
Comment (dstoup) [inline Patches/GTest/Patch.cmake]: Probably best to make this message STATUS, but I don't know how important that is in this context.

File: CMake/External_GTest.cmake (hunk 1 above)
Comment (mwoehlke-kitware) [inline CMake/External_GTest.cmake]: Oops... fixed locally, managed to avoid being committed. Thanks.

File: Patches/GTest/Patch.cmake (hunk 3 above)
Comment (mwoehlke-kitware) [inline Patches/GTest/Patch.cmake]: That may be, but there are several others that are also not `STATUS`... I vote for "leave it now, then fix all of them" or "fix all of them first and then I'll fix the PR also"...

File: CMake/External_GTest.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    ${COMMON_CMAKE_ARGS}
+  )
+
+set(GTEST_ROOT ${fletch_BUILD_INSTALL_PREFIX} CACHE STRING "")
+
+file(APPEND ${fletch_CONFIG_INPUT} "
+########################################
+# GTest
+########################################
+set(GTEST_ROOT \$\{fletch_ROOT\})
```
Comment (dstoup) [inline CMake/External_GTest.cmake]: I steered you a bit wrong the other day. I found that escaping the { } is not required and causes issues for CMake 2.8.12. Can you remove all escapes except the $ ... thanks!

**Comment by mwoehlke-kitware** [comment]
Is this still blocked? I'd like to start using this in kwiver... (See kitware/kwiver#277.)

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request merged**